tensorflow/core/kernels/mkl/mkl_tmp_ops.cc

```diff
@@ -21,34 +21,48 @@ limitations under the License.
 namespace tensorflow {
 
 // This file contains temporary registrations for some of the Eigen CPU backend
-// operators for BFloat16 type. The kernel registered for all these ops simply
-// raises an error. We do this so that MKL graph pass can rewrite these ops into
-// corresponding MKL ops. Without such registrations, Placer component in
-// TensorFlow fails because Eigen CPU backend does not support these ops in
-// BFloat16 type.
+// operators for Float32, BFloat16 and Half types. The kernels registered for
+// all these ops simply raise an error. We do this so that MKL graph pass can
+// rewrite these ops into corresponding MKL ops. Without such registrations,
+// Placer component in TensorFlow fails because Eigen CPU backend does not
+// support these ops for the above types.
 
 namespace {
-class RaiseBfloat16Error : public OpKernel {
+template <typename T>
+class RaiseError : public OpKernel {
  public:
-  explicit RaiseBfloat16Error(OpKernelConstruction* context)
-      : OpKernel(context) {
+  explicit RaiseError(OpKernelConstruction* context) : OpKernel(context) {
     OP_REQUIRES(context, false,
-                errors::InvalidArgument("Op does not support bfloat16 inputs"));
+                errors::InvalidArgument(absl::StrCat(
```

Comment:
Please use `absl::InvalidArgumentError` instead.
```suggestion
                absl::InvalidArgumentError(absl::StrCat(
```

Reply:
Fixed here and in other files modified by this PR.